examples/autocomplete.amp.html

```diff
@@ -0,0 +1,118 @@
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <script async custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-0.1.js"></script>
+  <script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
+  <script async custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js"></script>
+  <script async custom-element="amp-selector" src="https://cdn.ampproject.org/v0/amp-selector-0.1.js"></script>
+  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.1.js"></script>
+  <style amp-custom>
+  body {
+    padding: 10px;
+  }
+  .search-container {
+    display: flex;
+    flex-flow: row nowrap;
+  }
+  .search-box, .search-submit, .select-option {
+    font-size: 1.25em;
+  }
+  .autocomplete-box {
+    box-shadow: 0px 2px 6px rgba(0,0,0,.3);
+  }
+  .search-box {
+    flex: 4 0 0;
+    padding: 5px;
+    border-radius: 4px 0 0 4px;
+    border: 2px solid lightgray;
+    border-right: 0;
+  }
+  .search-submit {
+    flex: 1 0 0;
+    padding: 5px;
+    border-radius: 0 4px 4px 0;
+    border: 2px solid #4b68ff;
+    background: #4b68ff;
+    color: #fff;
+  }
+  .select-option {
+    display: block;
+    box-sizing: border-box;
+    height: 30px;
+    line-height: 30px;
+    padding-left: 10px;
+    background: #ddd;
+    outline: none!important;
+  }
+  .select-option:focus {
+    box-shadow: inset rgb(94, 158, 215) 0px 0px 45px;
+  }
+  .select-option:focus {
+    background: #fff;
+  }
+  .select-option:nth-child(2n) {
+    background: #eee;
+  }
+  </style>
+</head>
+<body>
+  <h1>Programming language search</h1>
+  <form
+    method="post"
+    action-xhr="/form/autocomplete/search"
+    target="_blank"
+    id="search-form"
+    on="submit:autocomplete-list.hide;submit-success:results.show"
+  >
+    <div class="search-container">
+      <input
+        id="query"
+        type="text"
+        class="search-box"
+        on="input-debounced:AMP.setState({
+          query: event.value,
+          autocomplete: event.value
+        }),autocomplete-list.show,results.hide"
+        name="term"
+        [value]="query"
+        autofocus/>
+      <button class="search-submit" type="submit">Search</button>
+    </div>
+    <amp-list
+      class="autocomplete-box"
+      layout="fixed-height"
+      height="120"
+      src="/form/autocomplete/query"
+      [src]="'/form/autocomplete/query?q=' + autocomplete"
```

Comment:
If `autocomplete` isn't defined in the first `AMP.setState` action, the src attribute will be changed to `'/form/autocomplete/query?q=null"`

I recommend changing this to

```HTML
 [src]="'/form/autocomplete/query' + (autocomplete ? '?q=' + autocomplete : '')
```

that way, null can't get into the query param and the default evaluation of the expression for `[src]` is equal to the initial vlaue for `src`.